src/snowflake/snowpark/mock/plan.py

```diff
@@ -641,6 +641,22 @@ def outer_join(base_df):
     if isinstance(source_plan, UnresolvedRelation):
         table_registry = analyzer.session._conn.table_registry
         return table_registry.read_table(source_plan.name)
+    if isinstance(source_plan, Sample):
+        res_df = execute_mock_plan(source_plan.child)
+
+        if source_plan.row_count and source_plan.row_count > 100000:
```

Comment:
Why do we need `source_plan.row_count`? Can it be `None`? Also, I wonder if we need to check for negative values here.

Reply:
It can be `None` if `source_plan.probability_fraction` is specified. I will add the negative check 👍